src/Testing/FakeNutgram.php

```diff
@@ -280,4 +283,48 @@ public function overrideMiddleware(string|array $middleware): self
 
         return $this;
     }
+
+    /**
+     * Get the actual data from the request.
+     * @param  Request  $request
+     * @param  array  $mapping
+     * @return array
+     * @throws JsonException
+     */
+    public static function getActualData(Request $request, array $mapping = []): array
+    {
+        //get content type
+        $contentType = $request->getHeaderLine('Content-Type');
+
+        //get body
+        $body = (string)$request->getBody();
+
+        //get data from json
+        if (str_contains($contentType, 'application/json')) {
+            return json_decode($body, true, flags: JSON_THROW_ON_ERROR);
+        }
+
+        //get data from form data
+        if (str_contains($contentType, 'multipart/form-data')) {
+            $formData = FormDataParser::parse($request);
+            $params = $formData->params;
+
+            //remap types lost in the form data parser
+            if (count($mapping) > 0) {
+                foreach ($params as $key => &$value) {
```

Comment:
Consider using an array_walk_recursive, modifying foreach while looping can be quite slow, and it doesn't handle nested structures